sycl/test/devicelib/math_test.cpp

```diff
@@ -9,64 +9,821 @@ namespace s = cl::sycl;
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLgamma>([=]() {
+        float a = NAN;
+        res_access[0] = lgammaf(a);
+      });
+    });
+  }
+
+  assert(std::isnan(result));
+}
+
+class DeviceLog1p;
+
+void device_log1p_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLog1p>([=]() {
+        float a = 0;
+        res_access[0] = log1pf(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceLog2;
+
+void device_log2_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLog2>([=]() {
+        float a = 1;
+        res_access[0] = log2f(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceLogb;
+
+void device_logb_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceLogb>([=]() {
+        float a = 1;
+        res_access[0] = logbf(a);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceRemainder;
+
+void device_remainder_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  float ref = 0.5;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceRemainder>([=]() {
+        float a = 0.5;
+        float b = 1;
+        res_access[0] = remainderf(a, b);
+      });
+    });
+  }
+
+  assert(result == ref);
+}
+
+class DeviceRemquo;
+
+void device_remquo_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{1};
+  float result = -1;
+  int quo = -1;
+  float ref = 0.5;
+  {
+    s::buffer<float, 1> buffer1(&result, numOfItems);
+    s::buffer<int, 1> buffer2(&quo, numOfItems);
+    deviceQueue.submit([&](cl::sycl::handler &cgh) {
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      auto quo_access = buffer2.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceRemquo>([=]() {
+        float a = 0.5;
+        float b = 1;
+        res_access[0] = remquof(a, b, &quo_access[0]);
+      });
+    });
+  }
+
+  assert(result == ref);
```

Comment:
Is this reliable way to validate math functions result?
NOTE: math functions are allowed to produce "inaccurate" results.